File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dict1 = {}
    answer = ''
    for str in completion:
        if str in dict1:
            dict1[str] += 1
        else:
            dict1[str] = 1
    
    for str in participant:
        if str not in dict1 or dict1[str] == 0:
            answer = str
            break
        dict1[str] -= 1
        
    return answer

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> completor = new HashMap<>();
        for(String complete : completion){
            completor.put(complete, completor.getOrDefault(complete, 0) + 1);
        }
        for(String join : participant){
            if(!completor.containsKey(join)){
                return join;
            }
            int a = completor.get(join);
            if(a == 0){
                return join;
            }else{
                completor.replace(join, a-1);
            }
        }
        return answer;
    }
}